### ISSUE TITLE
Add a way to reload a parser without restarting Neovim

### ISSUE THREAD

**Issue by darthdeus** [nvim-treesitter/nvim-treesitter#5328]
**Is your feature request related to a problem? Please describe.**
When I work on a treesitter grammar a new language I constantly restart vim in a second window on every change. It would be useful to have a command to "reload" the parser without having to quit Neovim.

**Describe the solution you'd like**
Add a `:TSReload lang` command.

**Describe alternatives you've considered**
I tried making this easier with sessions, but this still leaves the problem of a `:TSPlaygroundToggle` and having to close/reopen its buffer.

**Additional context**
TL;DR: Running `:TSInstallSync foo` requires neovim restart, would be useful for parser developers if it didn't.

**Comment by amaanq** [comment]
imo when developing a grammar you shouldn't need to be reinstalling it so often, you should ensure it's in a somewhat stable state before doing so *and then* try out queries

**Comment by lucario387** [comment]
Technically, all neovim does for loading a parser is

- Do a `vim._ts_add_language` to load the parser at the specific location defined
- Use the parser

For unloading, we have `vim._ts_remove_language`. Never intended for plugins and usage outside of neovim core though.

But it's not exposed explicity, so you'd need to do some digging if you really want to remove the parser and reload it.

Tl;dr: to create some parser reloading functionality, just do a `vim._ts_remove_language` then re add it using `vim._ts_add_language`. Results may not be what you want

**Comment by clason** [comment]
Agree; this is not safe nor reasonable to support officially. In any case, parser management is handled by Neovim, not nvim-treesitter, so this request is out of scope here.

**Comment by theHamsta** [comment]
When you do a `TSInstall` it will automatically unload the old grammar and load the new one. This is already implemented in nvim-treesitter. E.g. when you have broken queries, you will notice that `:TSUpdate` will load the new parser and make errors go away in the running nvim instance
